components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1354,6 +1319,49 @@ private static String getTenantUrl(String url, String tenantDomain) throws URISy
         }
     }
 
+    /**
+     * Builds a URL with a given context in both the tenant-qualified url supported mode and the legacy mode. Returns
+     * the absolute URL build from the default context in the tenant-qualified url supported mode and returns the
+     * absolute url build from file configuration context.
+     *
+     * @param defaultContext              Default URL context.
+     * @param getValueFromFileBasedConfig File-based Configuration.
+     * @return Absolute URL.
+     */
+    public static String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
+
+        String url = null;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                url = buildServiceUrl(defaultContext);
+            } catch (URLBuilderException e) {
+                throw new IdentityRuntimeException("Error while building url for context: " + defaultContext);
+            }
+        } else {
+            String oauth2EndpointURLInFile = null;
+            if (getValueFromFileBasedConfig != null) {
+                oauth2EndpointURLInFile = getValueFromFileBasedConfig.get();
+            }
```

Comment:
Rather doing the null check here, it would be better to have another method that does not take the getValueFromFileBasedConfig function.